CODEWALK.md

```diff
@@ -346,6 +346,118 @@ essentially who is storing what data, for what fee and which sectors have been s
 The keystore contains the binary encoded peer key for interacting securely over the network. 
 This data lives in a file at `$HOME/.filecoin/keystore/self`.
 
+## libp2p Networking
+
+As mentioned above, Filecoin relies on [libp2p](https://libp2p.io/) for its
+networking needs. Here we'll explain some of the mechanics that drive the
+network layer, in particular:
+
+1. Node discovery
+2. NAT topology determination
+3. Autorelay services for solving connectivity issues
+
+### Node discovery
```

Comment:
would you mind adding a sentence about:
- how my entry gets added to the routing table and when/how it gets pruned
- what happens if there are multiple entries for my peerid
- whether dht messages flow over an existing or separate physical connection. i think there are many logical streams that occur over the same physical connection

we should probably say what problems the dht solves for us as well. i think it is
1. ensure the network stays well-connect so that if some nodes go down it continues to operate
2. enable us to map from peerid to dialable multiaddr so that a client can connect to a miner whose peerid is found on chain in the miner actor

@whyrusleeping this are the two problems yes?

Reply:
Done.